From 8cfaa3737abf0cee7392403769120148ed4fe554 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:50 -0400
Subject: [ethernet] qlcnic: Turn on promiscous mode for SRIOV PF

Message-id: <1408626422-3941-67-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88597
O-Subject: [PATCH RHEL 7.1 066/138] qlcnic: Turn on promiscous mode for SRIOV PF.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 9eb0948af7ee41f29dcd7881f5087b596b46244c

o By default, SRIOV PF will have promiscous mode on.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index a4d6442..02de171 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -1618,6 +1618,10 @@ int qlcnic_83xx_nic_set_promisc(struct qlcnic_adapter *adapter, u32 mode)
 
  cmd->type = QLC_83XX_MBX_CMD_NO_WAIT;
  qlcnic_83xx_set_interface_id_promisc(adapter, &temp);
+
+ if (qlcnic_84xx_check(adapter) && qlcnic_sriov_pf_check(adapter))
+  mode = VPORT_MISS_MODE_ACCEPT_ALL;
+
  cmd->req.arg[1] = mode | temp;
  err = qlcnic_issue_cmd(adapter, cmd);
  if (!err)
-- 
1.7.1